components/MarkdownTextBlock/src/TextElements/MyImage.cs

```diff
@@ -0,0 +1,158 @@
+        _imageProvider = config?.ImageProvider;
+        _svgRenderer = config?.SVGRenderer == null ? new DefaultSVGRenderer() : config.SVGRenderer;
+        Init();
+        int.TryParse(
+            htmlNode.GetAttributeValue("width", "0"),
+            NumberStyles.Integer,
+            CultureInfo.InvariantCulture,
+            out var width
+        );
+        int.TryParse(
+            htmlNode.GetAttributeValue("height", "0"),
+            NumberStyles.Integer,
+            CultureInfo.InvariantCulture,
+            out var height
+        );
+        if (width > 0)
+        {
+            _precedentWidth = width;
+        }
+        if (height > 0)
+        {
+            _precedentHeight = height;
+        }
+    }
+
+    private void Init()
+    {
+        _image.Loaded += LoadImage;
+        _container.Child = _image;
+    }
+
+    private async void LoadImage(object sender, RoutedEventArgs e)
+    {
+        if (_loaded) return;
+        try
+        {
+            if (_imageProvider != null && _imageProvider.ShouldUseThisProvider(_uri.AbsoluteUri))
+            {
+                _image = await _imageProvider.GetImage(_uri.AbsoluteUri);
+                _container.Child = _image;
+            }
+            else
+            {
+                HttpClient client = new HttpClient();
+
+                // Download data from URL
+                HttpResponseMessage response = await client.GetAsync(_uri);
+
+
+                // Get the Content-Type header
+#pragma warning disable CS8600 // Converting null literal or possible null value to non-nullable type.
+#pragma warning disable CS8602 // Dereference of a possibly null reference.
+                string contentType = response.Content.Headers.ContentType.MediaType;
+#pragma warning restore CS8602 // Dereference of a possibly null reference.
+#pragma warning restore CS8600 // Converting null literal or possible null value to non-nullable type.
+
+                if (contentType == "image/svg+xml")
+                {
+                    var svgString = await response.Content.ReadAsStringAsync();
+                    var resImage = await _svgRenderer.SvgToImage(svgString);
+                    if (resImage != null)
+                    {
+                        _image = resImage;
+                        _container.Child = _image;
+                    }
+                }
+                else
+                {
+                    byte[] data = await response.Content.ReadAsByteArrayAsync();
+                    // Create a BitmapImage for other supported formats
+                    BitmapImage bitmap = new BitmapImage();
+                    using (InMemoryRandomAccessStream stream = new InMemoryRandomAccessStream())
+                    {
+                        // Write the data to the stream
+                        await stream.WriteAsync(data.AsBuffer());
+                        stream.Seek(0);
+
+                        // Set the source of the BitmapImage
```

Comment:
I think you can directly read the response as a stream instead?

Reply:
> I think you can directly read the response as a stream instead then set it as an image source?

Some images are dynamic in nature. So I guess that won't work.

Reply:
no going to change this one for now.